R/calculate_within_batch_ari.R

```diff
@@ -0,0 +1,94 @@
+#' Calculate within-batch ARI scores from an integrated SCE object.
+#'
+#'
+#' @param individual_sce_list A list of individual SCE objects
+#' @param individual_pc_name The name that allows access to the PCs in the
+#'  individual SCE objects. Default is "PCA".
+#' @param integrated_pc_name The name that allows access to the PCs in the
+#'  integrated SCE object.  Example: "fastMNN_PCA".
+#' @param integrated_sce The integrated SCE object
+#' @param batch_column The variable in `integrated_sce` indicating the grouping of interest.
+#'  Generally this is either batches or cell types. Default is "library_id".
+#' @param seed Seed for initializing random sampling
+#'
+#' @return Tibble with three columns: `ari`, representing the calculated ARI values;
+#'   `batch_id`, the batch id associated with each `ari`; `pc_name`, the name
+#'   associated with the pc results
+#'
+#' @import SingleCellExperiment
+#' @importFrom rlang .data
+#'
+#' @export
+calculate_within_batch_ari <- function(individual_sce_list,
+                                       individual_pc_name,
+                                       integrated_pc_name,
+                                       integrated_sce,
+                                       batch_column = "library_id",
+                                       seed = NULL) {
+  # Set the seed for subsampling and clustering
+  set.seed(seed)
+
+  # Check that list of SCE objects is named
+  batch_ids <- names(individual_sce_list)
+  if (is.null(batch_ids)) {
+    stop("Must provide a named list of SCE objects.")
+  } else {
+    # Make sure the batch ids provided match between the list and the integrated object
+    if (!(identical(
+      sort(batch_ids),
+      sort(unique(colData(integrated_sce)[, batch_column]))
+    ))
+    ) {
+      stop("Names of provided SCE objects included in the individual SCE object
+           do not match batch IDs present in the batch_column of the integrated object")
+    }
+  }
+
+  # Pull out the PCs or analogous reduction from integrated object
+  integrated_pcs <- reducedDim(integrated_sce, integrated_pc_name)
+
+  # Cluster integrated pcs only one time
+  integrated_clustering_result <- integrated_sce |>
```

Comment:
```suggestion
  merged_sce <- merged_sce |>
```
I don't think we need to save an entirely new object. This function is adding in a new column `integrated_clusters` to an existing SCE so we can use the same object. I would then later pull out the cluster column so it's easier to follow what's happening. 
This change depends on you taking my suggestion to change to `merged_sce`.